### PR TITLE
Add job run cleaner

### PR DESCRIPTION
**Pull request by johanbrandhorst**
HCP Boundary reports that the job_runs table is filling up with data, and this is something that will be affecting customers too. Create a cleaner job that cleans up old job runs.

### CODE REVIEW

File: internal/daemon/controller/controller.go [hunk 1]
```diff
@@ -28,8 +28,8 @@ import (
 	kmsjob "github.com/hashicorp/boundary/internal/kms/job"
 	"github.com/hashicorp/boundary/internal/observability/event"
 	"github.com/hashicorp/boundary/internal/plugin/host"
-	hostplugin "github.com/hashicorp/boundary/internal/plugin/host"
```
Comment (louisruch) [inline internal/daemon/controller/controller.go:31]: Nice catch

File: internal/db/schema/migrations/oss/postgres/62/01_job_run_index.up.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+begin;
+
+  delete from job_run where status='completed';
+
+  create index on job_run (status);
```
Comment (louisruch) [inline internal/db/schema/migrations/oss/postgres/62/01_job_run_index.up.sql:5]: Most of the index's we create, apart from 1 that I found create a named index. https://github.com/hashicorp/boundary/blob/main/internal/db/schema/migrations/oss/postgres/50/01_session_list_index.up.sql Often we also add a comment on the index https://github.com/hashicorp/boundary/blob/main/internal/db/schema/migrations/oss/postgres/10/04_vault_credential.up.sql#L182

File: internal/scheduler/cleaner/cleaner_job.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+func newCleanerJob(w db.Writer) *cleanerJob {
+	return &cleanerJob{
+		w: w,
+	}
+}
+
+// Status reports the job’s current status. We never change these values as
+// this job never finishes.
```
Comment (louisruch) [inline internal/scheduler/cleaner/cleaner_job.go:23]: Not sure I follow the `this job never finishes` comment. Though I am fine with not reporting status here

File: internal/scheduler/cleaner/cleaner.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/scheduler"
+)
+
+// RegisterJob registers the cleaner job with the provided scheduler.
+func RegisterJob(ctx context.Context, s *scheduler.Scheduler, w db.Writer) error {
```
Comment (louisruch) [inline internal/scheduler/cleaner/cleaner.go:12]: This func is never tested

File: internal/scheduler/cleaner/cleaner.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	const op = "cleaner.RegisterJob"
+	if s == nil {
```
Comment (louisruch) [inline internal/scheduler/cleaner/cleaner.go:14]: Might want to nil check the writer

**Review by louisruch — COMMENTED** [review]
Thanks for getting onto this!

File: internal/db/schema/migrations/oss/postgres/62/01_job_run_index.up.sql (hunk 2 above)
Comment (johanbrandhorst) [reply internal/db/schema/migrations/oss/postgres/62/01_job_run_index.up.sql]: Good shout, I added a comment and a name.

File: internal/scheduler/cleaner/cleaner_job.go (hunk 3 above)
Comment (johanbrandhorst) [reply internal/scheduler/cleaner/cleaner_job.go]: I removed this part of the comment, it was just confusing.

File: internal/scheduler/cleaner/cleaner.go (hunk 4 above)
Comment (johanbrandhorst) [reply internal/scheduler/cleaner/cleaner.go:13]: I added a test for this, thanks!

File: internal/scheduler/cleaner/cleaner.go (hunk 5 above)
Comment (johanbrandhorst) [reply internal/scheduler/cleaner/cleaner.go:15]: Good idea, done!

**Review by louisruch — APPROVED** [review]
:shipit:

**Pull request merged**